#! /bin/sh /usr/share/dpatch/dpatch-run
## default-ripcache-size-auto.dpatch by Till Kamppeter <till.kamppeter@gmail.com>
##
## DP: Set the default RIPCache to "auto". This makes Ghostscript
## DP: (pstoraster, pdftoraster) determine the needed amount of memory
## DP: completely automatically, which makes nearly all files rendering
## DP: correctly with the CUPS Raster output device. This requires Ghostscript
## DP: 9.00 or newer or Ghostscript with the appropriate change patched in.
## DP: The upstream hardcoded default of 8 MB (default of CUPS and of
## DP: Ghostscript 8.71 and older) is too small for complex documents.
## DP: The RIPCache is also used by the imagetops CUPS filter, which will
## DP: default to 32 MB with our "auto" setting. It worked already perfectly
## DP: with the old 8 MB.

@DPATCH@
diff -urNad cups-1.4.1~/scheduler/conf.c cups-1.4.1/scheduler/conf.c
--- cups-1.4.1~/scheduler/conf.c	2009-09-23 22:58:51.678030419 +0200
+++ cups-1.4.1/scheduler/conf.c	2009-09-23 22:59:03.426053168 +0200
@@ -546,7 +546,7 @@
 
   cupsdClearString(&DefaultPaperSize);
 
-  cupsdSetString(&RIPCache, "8m");
+  cupsdSetString(&RIPCache, "auto");
 
   cupsdSetString(&TempDir, NULL);
 
diff -urNad cups-1.4.1~/scheduler/cupsfilter.c cups-1.4.1/scheduler/cupsfilter.c
--- cups-1.4.1~/scheduler/cupsfilter.c	2009-06-05 23:38:52.000000000 +0200
+++ cups-1.4.1/scheduler/cupsfilter.c	2009-09-23 22:58:57.158032299 +0200
@@ -1218,7 +1218,7 @@
   else
     set_string(&FontPath, CUPS_FONTPATH);
 
-  set_string(&RIPCache, "8m");
+  set_string(&RIPCache, "auto");
 
   if ((temp = getenv("CUPS_SERVERBIN")) != NULL)
     set_string(&ServerBin, temp);
